File: app/src/main/java/io/mokshjn/cosmo/playback/Playback.java
package io.mokshjn.cosmo.playback;

import android.support.v4.media.session.MediaSessionCompat;

/**
 * Created by moksh on 19/3/17.
 */

public interface Playback {
    void start();

    void stop(boolean notifyListeners);

    int getState();

    void setState(int state);

    boolean isConnected();

    boolean isPlaying();

    int getCurrentStreamPosition();

    void setCurrentStreamPosition(int pos);

    void updateLastKnownStreamPosition();

    void play(MediaSessionCompat.QueueItem item);

    void pause();

    void seekTo(int position);

    String getCurrentMediaId();

    void setCurrentMediaId(String mediaId);

    void setCallback(Callback callback);

    interface Callback {

        void onCompletion();

        void onPlaybackStatusChanged(int state);

        void onError(String error);

        void setCurrentMediaId(String mediaId);
    }
}

File: app/src/main/java/io/mokshjn/cosmo/playback/Playback.kt
package io.mokshjn.cosmo.playback

/**
 * Created by moksh on 27/11/17.
 */
interface Playback {
    fun setDataSource(path: String): Boolean

    fun setNextDataSource(path: String?)

    fun setCallbacks(callbacks: PlaybackCallbacks)

    fun isInitialized(): Boolean

    fun start(): Boolean

    fun stop()

    fun release()

    fun pause(): Boolean

    fun isPlaying(): Boolean

    fun duration(): Int

    fun position(): Int

    fun seek(to: Int): Int

    fun setVolume(volume: Float): Boolean

    fun setAudioSessionId(sessionId: Int): Boolean

    fun getAudioSessionId(): Int

    interface PlaybackCallbacks {
        fun onTrackWentToNext()
        fun onTrackEnded()
    }
}